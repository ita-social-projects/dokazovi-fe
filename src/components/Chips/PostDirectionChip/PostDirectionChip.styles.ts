import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  () => ({
    directionChip: () => ({
      backgroundColor: '#FFFFFF',
      borderRadius: '300px',
      padding: '15px 8px 15px 8px',
      boxShadow: '0px 1px 2px rgba(0, 0, 0, 0.08)',
      margin: '0px 15px 15px 0px',
    }),
  }),
  { name: 'PostDirectionChip' },
);
