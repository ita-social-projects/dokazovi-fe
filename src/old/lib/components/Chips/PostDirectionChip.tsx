import React from 'react';
import { makeStyles, Chip } from '@material-ui/core';

export interface IPostDirectionChipProps {
  labelName?: string;
  backgroundColor?: string;
  handleClick?: () => void;
}

const useStyles = makeStyles(
  () => ({
    directionChip: (props: IPostDirectionChipProps) => ({
      backgroundColor: props.backgroundColor,
      borderRadius: '15px',
      padding: '0px 8px 0px 8px',
    }),
  }),
  { name: 'PostDirectionChip' },
);

const PostDirectionChip: React.FC<IPostDirectionChipProps> = (props) => {
  const classes = useStyles(props);
  const { labelName, handleClick } = props;

  const onClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    e.stopPropagation();
    if (handleClick) {
      handleClick();
    }
  };

  return (
    <Chip
      className={classes.directionChip}
      label={labelName}
      size="small"
      onClick={onClick}
    />
  );
};

export default PostDirectionChip;
