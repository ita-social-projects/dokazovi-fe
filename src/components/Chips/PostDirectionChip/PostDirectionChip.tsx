/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-shadow */
import React from 'react';
import { Chip } from '@material-ui/core';
// import { DoneIcon } from '../icons/DoneIcon';
import { useStyles } from './PostDirectionChip.styles';

export interface IPostDirectionChipProps {
  labelName?: string;
  backgroundColor?: string;
  handleClick?: () => void;
  handleDelete?: (arg0: string | undefined) => void;
}

const PostDirectionChip: React.FC<IPostDirectionChipProps> = (props) => {
  const classes = useStyles(props);
  const { labelName, handleClick, handleDelete } = props;

  const onClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    if (handleClick) {
      handleClick();
    }
  };

  const onDeleteClick = (labelName: string | undefined) => {
    if (handleDelete) {
      handleDelete(labelName);
    }
  };

  return (
    <>
      <Chip
        key={labelName}
        className={classes.directionChip}
        label={labelName}
        size="small"
        onClick={onClick}
        onDelete={() => onDeleteClick(labelName)}
      />
    </>
  );
};

export default PostDirectionChip;
