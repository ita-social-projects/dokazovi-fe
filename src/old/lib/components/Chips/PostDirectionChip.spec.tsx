import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PostDirectionChip, {
  IPostDirectionChipProps,
} from './PostDirectionChip';

const mocks: IPostDirectionChipProps = {
  labelName: 'Test Label Name',
};

describe('PostDirectionChip', () => {
  it('should render correctly when loading status is pending', () => {
    const { asFragment } = render(
      <PostDirectionChip labelName={mocks.labelName} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display correct label name', () => {
    render(<PostDirectionChip labelName={mocks.labelName} />);

    expect(screen.getByText('Test Label Name')).toBeInTheDocument();
  });
});

describe('PostDirectionChip Events', () => {
  it('should handle click on it by executing given function', () => {
    const handleClick = jest.fn();
    render(
      <PostDirectionChip
        labelName={mocks.labelName}
        handleClick={handleClick}
      />,
    );

    const label = screen.getByText('Test Label Name');
    userEvent.click(label);

    expect(handleClick).toBeCalledTimes(1);
  });
});
