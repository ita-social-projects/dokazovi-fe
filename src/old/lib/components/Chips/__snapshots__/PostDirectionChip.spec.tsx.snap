// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDirectionChip should render correctly when loading status is pending 1`] = `
<DocumentFragment>
  <div
    class="MuiButtonBase-root MuiChip-root PostDirectionChip-directionChip-1 PostDirectionChip-directionChip-2 MuiChip-sizeSmall MuiChip-clickable"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelSmall"
    >
      Test Label Name
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</DocumentFragment>
`;
